Override the plugin URL found in update-center.json with JENKINS_PLUGINS_URL env var value (#277)
jenkinsci/plugin-installation-manager-tool: README.md

* `JENKINS_UC_DOWNLOAD`: used to configure the URL from where plugins will be downloaded from. Often used to cache or to proxy the Jenkins plugin download site. 
If set then all plugins will be downloaded through that URL.
  

> * `JENKINS_PLUGINS_URL`: Rewrites the plugin url to the specified value before a plugin is downloaded. Use this option to download from a specific server and bypass the mirrors and force the use of your own server. For example, if your server has the Jenkins plugins at `https://company.com/jenkins-plugins` then simply set this variable to that value. From the specified URL, the plugin must be found under `pluginName/version/pluginName.hpii`. There is no need to re-write the `update-center.json` file as this feature will re-write the URL right before the plugin is fetched. This is useful when the mirrors are broken or if you have an internal policy requirement. You still need access to the official `update-center.json` on the Jenkins servers file.

Reviewer: ```suggestion
* `JENKINS_PLUGINS_URL`: Rewrites the plugin url to the specified value before a plugin is downloaded. Use this option to download from a specific server and bypass the mirrors and force the use of your own server. For example, if your server has the Jenkins plugins at `https://company.com/jenkins-plugins` then simply set this variable to that value. From the specified URL, the plugin must be found under `pluginName/version/pluginName.hpi`. There is no need to re-write the `update-center.json` file as this feature will re-write the URL right before the plugin is fetched. This is useful when the mirrors are broken or if you have an internal policy requirement. You still need access to the official `update-center.json` in the Jenkins upda center.
```

Author: @oleg-nenashev I think `JENKINS_UC_DOWNLOAD` requires the `update-center.json` to be hosted on a server as well. `JENKINS_PLUGINS_URL` does not. So a user could continue to fetch the update-center.json from the official server while at the same time use another server for the plugins. I do not know if this is a common occurence, but it is certainly mine.

A use case might clarify:
```
export JENKINS_PLUGINS_URL=https://server.com/customPath
java -jar jenkins-plugin-manager.jar \
    --plugin-file plugins.yml \
    --plugin-download-directory plugins \
    --jenkins-version 2.235.3
```

And thanks for the quick review!

Reviewer 2: Jenkins uc download shouldn’t require that, the point of it is to just override the download url

Author: Ah, you're right, I see what I did. I messed up the documentation.
